src/V3LinkLValue.cpp

```diff
@@ -45,7 +45,7 @@ class LinkLValueVisitor final : public VNVisitor {
     // Result handing
     void visit(AstNodeVarRef* nodep) override {
         // VarRef: LValue its reference
-        if (m_setIfRand && !(nodep->varp() && nodep->varp()->isRand())) return;
+        if (m_setIfRand && !(nodep->varp() && nodep->varp()->rand().isRand())) return;
```

Comment:
There's a lot of these changes, suggest revert them and add 'AstVar::isRand() const { return rand().isRand; }'